eth_utils/abi.py

```diff
@@ -50,6 +80,188 @@ def _abi_to_signature(abi: Dict[str, Any]) -> str:
     return function_signature
 
 
+def _filter_by_type(_type: str, contract_abi: ABI) -> List[ABIElement]:
+    return [abi for abi in contract_abi if abi["type"] == _type]
+
+
+def _filter_by_name(name: str, contract_abi: ABI) -> List[ABIElement]:
+    return [
+        abi
+        for abi in contract_abi
+        if (
+            abi["type"] not in ("fallback", "constructor", "receive")
+            and abi["name"] == name
+        )
+    ]
+
+
+def _filter_by_argument_name(
+    argument_names: Collection[str], contract_abi: ABI
+) -> List[ABIElement]:
+    return [
+        abi
+        for abi in contract_abi
+        if set(argument_names).intersection(get_abi_input_names(abi))
+        == set(argument_names)
+    ]
+
+
+def _log_entry_data_to_bytes(
+    log_entry_data: Union[Primitives, HexStr, str],
+) -> HexBytes:
+    return hexstr_if_str(to_bytes, log_entry_data)
+
+
+def get_event_log_topics(
+    event_abi: ABIEvent,
+    topics: Optional[Sequence[HexBytes]] = None,
+) -> Sequence[HexBytes]:
+    """
+    Return topics from an event ABI.
+
+    :param event_abi: Event ABI.
+    :param type: `ABIEvent`
+    :param topics: Transaction topics from a `LogReceipt`.
+    :param type: `list[HexBytes]`
+    :return: Event topics from the event ABI.
+    :rtype: `list[HexBytes]`
+    """
+    if topics is None:
+        topics = []
+
+    if event_abi["anonymous"]:
+        return topics
+    elif len(topics) == 0:
+        # raise MismatchedABI("Expected non-anonymous event to have 1 or more topics")
```

Comment:
Cant import this from web3, so should this be declared in exceptions.py and then all of web3 can point at the new one here in eth-utils?

Reply:
I think that seems reasonable, yeah. We wouldn't want to import this from web3.py anyhow since that is a `Web3Exception`, specifically only meant to be raised inside of web3.py. 

Then, in *web3.py,* we can likely do:

```py
from eth_utils import MismatchedABI as EthUtilsMismatchedABI

class MismatchedABI(Web3Exception, EthUtilsMismatchedABI)
    ...
```